src/ansys/fluent/core/solver/flobject.py

```diff
@@ -372,17 +388,49 @@ class String(SettingsBase[str], Textual):
     _state_type = str
 
 
-class Filename(SettingsBase[str], Textual):
+class InputOutputFileBase:
+    def is_input(self):
+        if self.file_purpose() == "input":
+            return True
+        return False
+
+    def before_execute(self, value):
+        try:
+            self.remote_file_handler.upload(file_name=value)
```

Comment:
We're leaving it to the caller decide whether upload/download is appropriate. That isn't what was discussed.

Reply:
We are doing upload before command execution and download after command execution by checking whether file_purpose is input/output therefore I have implemented  self.remote_file_handler.upload in before_execute() and self.remote_file_handler.download in after_execute.